
syntax = "proto3";

package SnapProto;

message ReadDef {
    string bases = 1;
    int32 length = 2;
    string qualities = 3;
    string meta = 4;
}

message SingleResultDef {
    enum AlignmentResult {
        NOT_FOUND = 0;
        SINGLE_HIT = 1;
        MULTIPLE_HITS = 2;
        UNKNOWN_ALIGNMENT = 3;
    }

    AlignmentResult result = 1;
    int64 genomeLocation = 2;
    int32 score = 3;
    int32 mapq = 4;
    int32 direction = 5;
}

message AlignmentDef {
    repeated SingleResultDef results = 1;
    bool firstIsPrimary = 2;
    ReadDef read = 3;
}

